specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/cloudhsm.json

```diff
@@ -0,0 +1,1430 @@
+        }
+      ]
+    },
+    "BackupRestoreRequestBaseProperties": {
+      "type": "object",
+      "description": "Backup and Restore Common properties",
+      "properties": {
+        "azureStorageResourceUri": {
+          "type": "string",
+          "description": "The Azure blob storage container Uri which contains the backup"
+        },
+        "token": {
+          "type": "string",
+          "description": "The SAS token pointing to an Azure blob storage container. This property is reserved for Azure Backup Service."
+        }
+      },
+      "required": [
+        "azureStorageResourceUri"
+      ]
+    },
+    "BackupResponse": {
+      "type": "object",
+      "properties": {
+        "azureStorageResourceUri": {
+          "type": "string",
+          "description": "The Azure blob storage container Uri which contains the backup"
+        },
+        "backupId": {
+          "type": "string",
+          "description": "The ID of the backup."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRestoreBaseResponse"
+        }
+      ],
+      "description": "Backup operation Result"
+    },
+    "RestoreResponse": {
+      "type": "object",
+      "description": "Restore operation properties",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRestoreBaseResponse"
+        }
+      ]
+    },
+    "BackupRestoreBaseResponse": {
+      "type": "object",
+      "properties": {
+        "status": {
+          "type": "string",
+          "description": "Status of the backup operation"
+        },
+        "statusDetails": {
+          "type": "string",
+          "description": "The status details of backup operation"
+        },
+        "error": {
+          "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorDetail",
+          "description": "Error encountered, if any, during the backup operation."
+        },
+        "startTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "The start time of the backup operation in UTC"
+        },
+        "endTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "The end time of the backup operation in UTC",
+          "x-nullable": true
+        },
+        "jobId": {
+          "type": "string",
+          "description": "Identifier for the backup operation."
+        }
+      },
+      "description": "Backup and Restore operation common properties"
+    }
+  },
+  "parameters": {
+    "cloudHsmClusterName": {
+      "name": "cloudHsmClusterName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the Cloud HSM Cluster within the specified resource group. Cloud HSM Cluster names must be between 3 and 24 characters in length.",
```

Comment:
Did we ever validate if 24 chars supported?

Reply:
its a mistake in the description. allowed length is 23. updated description.